src/js/api/errors.ts

```diff
@@ -161,18 +165,23 @@ export const MSG_TECHNICAL_ERROR = 'technicalError';
  * @param {sourceError} [Error] - The lower level error, caught by the player, which resulted in this error.
  */
 export class PlayerError {
-    constructor(key, code, sourceError = null) {
+    code: ErrorCode;
+    sourceError: Error | PlayerError | null;
+    key: ErrorKey | null = null;
+    message: string | null = null;
+
+    constructor(key: ErrorKey | null, code: ErrorCode, sourceError: Error | PlayerError | null = null) {
```

Comment:
`key?: ErrorKey, code:...`

Reply:
since code is required, can't have optional params before. seems like we usually pass in null for our usages.